iModelCore/iModelPlatform/DgnCore/DgnMaterial.cpp

```diff
@@ -13,6 +13,28 @@ namespace dgn_ElementHandler
 
 END_BENTLEY_DGNPLATFORM_NAMESPACE
 
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+void RenderMaterial::_OnLoadedJsonProperties()
+    {
+        BeJsValue materialAssets = GetMaterialAssetsR();
+        if (materialAssets.hasMember("renderMaterial") && materialAssets["renderMaterial"].hasMember("Map"))
+            {
+            BeJsValue map = materialAssets["renderMaterial"]["Map"];
+            map.ForEachProperty([&](Utf8CP memberName, BeJsConst memberJson)
+                {
+                if (memberJson.isNumericMember("TextureId")) 
+                    {
+                    // Fix IDs that were previously stored as 64-bit integers rather than as ID strings.
+                    auto textureId = memberJson["TextureId"].GetId64<DgnTextureId>();

```

Comment:
If we get an invalid id (0) here I think we should log and maybe assert.  While this shouldn't happen with the current problem we should be defensive so we find other issues out before they hit us in a transformation workflow or something like that

Reply:
https://github.com/iTwin/imodel-native/pull/820/commits/d5f65fa7dd3c1aa3cd60a71fe942b1c3666f4d5e
How does this look?

Reply:
Use [PRIu64](https://stackoverflow.com/questions/16859500/what-is-priu64-in-c) to format unsigned 64-bit integers.

Reply:
Or, make life easy and use `GetElementId().ToHexStr()` and `%s`.

Reply:
switched to ToHexStr() and %s

Reply:
LGTM will let @ColinKerr resolve.